resources/uco_design_document.md

```diff
@@ -181,7 +213,35 @@ By default the UUIDs used in object identifiers should be randomly generated UUI
+     * This is also true for defining particular subclasses of existing `UcoInherentCharacterizationThing` classes
+* In some cases (e.g., Identity) (where purpose-specific object properties may not be appropriate) this also makes it more complicated for particular adopters (e.g., application subdomain ontologies or even specific communities or organizations) to scope their adoption to specific `UcoInherentCharacterizationThing` class-based characteristics.
+  * The purpose-specific object property approach requires them to define their own specialized complex shapes (defining property shapes for all of the separate purpose-specific object properties) for the relevant UcoObjects.
+* In some cases (e.g., Location) (where purpose-specific object properties may not be appropriate) this also makes it unnecessarily more complicated for expressing the same basic concept characterization using different mechanisms (e.g., a location expressed utilizing different coordinate systems).
+* In some cases where specific context may not yet be understood and more general property association is necessary (e.g. duck typing), the purpose-specific object properties approach is impractical.
+* In some cases where differing levels of detail may be needed to characterize a given concept and handling it all through subclassing involves impractical complexity. This can be seen in the UCO approach for observable objects due to lessons learned in the Cyber Observable eXpression (CybOX) standardization effort which is the base origin for the UCO approach. CybOX found it highly detrimental to attempt to manage multiple levels of characterization detail through purpose-specific properties on complex subclass hierarchies (e.g., a plain file, a plain raster image file, a raster image file with EXIF data, etc.). Similar lessons were also learned in the Structured Threat Information eXpression (STIX) standardization effort.
+
+
+Because of these many limitations/downsides and necessary use cases there is a need for a generalized object property mechanism (in addition to the common default purpose-specific approach) to associate some `UcoInherentCharacterizationThing` classes with `UcoObject` classes in cases where it is appropriate.
+
+Such a mechanism is currently implemented in UCO using the `Facet` subclass of `UcoInherentCharacterizationThing` class combined with the `hasFacet` object property on the `UcoObject` class which very significantly mitigates the above limitations/downsides as well as supports many of the necessary use cases.
+   
+A `Facet` is simply a `UcoInherentCharacterizationThing` that is associated with a `UcoObject` through use of a generic (`hasFacet`) rather than purpose-specific object property.
+Facets are fully consistent with ontological principles.

```

Comment:
@sbarnum , could you or @plbt5 please cite an ontological principles source?  Both you have waved hands at this phrase with great vigor in the past.  What I recall most from Paul is the principle of "Ontologies model reality, which is different from modeling data."

It might or might not be helpful to have a source noted for us all to identify common foundations.  But, the frequent allusion without substantiation has not progressed this discussion.